core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -42,6 +44,19 @@ object Consumer {
      */
     def shutdown(): Future[Done]
 
+    /**
+     * Stop producing messages from the `Source`, wait for stream completion
+     * and shut down the consumer `Source` so that all consumed messages
+     * reach the end of the stream.
+     * Failures in stream completion will be propagated, the source will be shut down anyway.
+     */
+    def drainAndShutdown[S](streamCompletion: Future[S])(implicit ec: ExecutionContext): Future[S] =
+      stop()
+        .flatMap(_ => streamCompletion)
+        .transformWith { _ =>
+          shutdown().flatMap(_ => streamCompletion)
+        }
```

Comment:
If the `stop()` future fails, then that error is going to be dropped. What about:

```
.transformWith { t =>
  shutdown().flatMap(_ => Future.fromTry(t))
}
```